src/firmware/main/app/primitives/primitive_manager.c

```diff
@@ -0,0 +1,187 @@
+#include "primitive_manager.h"
+
+#ifdef __arm__
+// TODO: do we need "FreeRTOS.h" here?
+#include <FreeRTOS.h>
+#include <semphr.h>
+#elif __unix__
```

Comment:
Maybe a quick comment for this file explaining why the typedefs are needed would be useful

Reply:
added:
```
// There are different semaphore implementations depending on if we're on a x86 or arm
// system, and we use typdefs here to switch between them
```